### ISSUE TITLE
defaults need revising

### ISSUE THREAD

**Issue by vcunat** [DNSSEC-Tools/DNSSEC-Tools#16]
Apparently [the defaults](https://github.com/DNSSEC-Tools/DNSSEC-Tools/blob/cc97499e53f1e473b5d2adf88d0bb6aa1fa467c1/dnssec-tools/tools/modules/defaults.pm#L37) haven't been touched since 2014, and I believe they need revising.  My conviction is that good default behavior in tools is essential for DNSSEC to succeed.

Notable points on a quick glance:
- `'algorithm'	   => "rsasha1",	# Encryption algorithm.`
  No please.  [Standards](https://tools.ietf.org/html/rfc8624#section-3.1) say "not recommended", but I'd personally be even stricter nowadays and certainly wouldn't allow it as default.
- NSEC3 set up seems too wasteful.
  ```
  'nsec3iter'	   => 100,              # default NSEC3 iterations
  'nsec3salt'	   => "random:64",      # default NSEC3 salt
  ```
  Iterations would better be kept to minimum, perhaps even zero.  (I just hope that this doesn't imply 64-**byte** salt.)
- 1024-bit RSA isn't considered very secure nowadays.  I'd personally suggest `ECDSAP256SHA256` as default (shorter packets, etc.), but I expect even 1280-bit RSA is still acceptable.

**Comment by hardaker** [comment]
All good points, thank you.  I think the salt length is bits not bytes.  I'll make some changes.

**Comment by vcunat** [comment]
64 bits sound a good compromise to me.  (Some people prefer to avoid salt altogether.)

**Comment by vcunat** [comment]
I think this was immediately solved in 013944e3365c6e57d0520edea4135d2cbca1a0df and its parent.
